### PR TITLE
L1t updates 75x

### PR DESCRIPTION
**Pull request by mulhearn**
Updates for L1 Trigger needed for 75X.  Updates from HI.  Calorimeter trigger updates.  Unpacker updates to improve data/hardware agreement.  Updates for Stage2 calorimeter.

### CODE REVIEW

**Comment by mulhearn** [comment]
Something went wrong with the conflict resolution.  Will update...

**Comment by mulhearn** [comment]
Fixed... there was a failed manual merge some (unused by standard sequences) Stage 2 code.

**Comment by mulhearn** [comment]
There is a corresponding PR in cms-data already:
https://github.com/cms-data/L1Trigger-L1TCalorimeter/pull/3
This is a small data file, but I can delete it once this PR is merged.

**Comment by mulhearn** [comment]
+1

**Comment by mulhearn** [comment]
please retest

**Comment by mulhearn** [comment]
+1

**Comment by mulhearn** [comment]
please test

**Comment by Martin-Grunewald** [comment]
I believe this PR broke HLT tests - see here:
https://cmssdt.cern.ch/SDT/cgi-bin//showAddOnLogs.py/slc6_amd64_gcc491/www/sat/7.5-sat-23/CMSSW_7_5_X_2015-05-23-2300/addOnTests/
Could the expert please have a look and advise?
Thanks!

**Comment by Martin-Grunewald** [comment]
@mulhearn @jimbrooke 
I believe this PR broke HLT tests - see here:
https://cmssdt.cern.ch/SDT/cgi-bin//showAddOnLogs.py/slc6_amd64_gcc491/www/sat/7.5-sat-23/CMSSW_7_5_X_2015-05-23-2300/addOnTests/
Could you please have a look and advise?
Thanks!

**Comment by Martin-Grunewald** [comment]
@mulhearn @jimbrooke @puigh @davidlange6 
I just checked explicitly: CMSSW_7_5_X_2015-05-23-1100 alone is fine, 
CMSSW_7_5_X_2015-05-23-1100 + this PR breaks the tests!
(CMSSW_7_5_X_2015-05-23-1100 is the most-recent IB before this PR was integated).

File: L1Trigger/L1TCommon/python/customsPostLS1.py [hunk 1]
```diff
@@ -94,7 +95,16 @@ def customiseSimL1EmulatorForPostLS1_HI(process):
     # load the Stage 1 configuration
     process = customiseSimL1EmulatorForStage1(process)
     # set the Stage 1 heavy ions-specific parameters
-    process.simCaloStage1Digis.FirmwareVersion = cms.uint32(1)
+    # all of these should eventually end up in a GT
+    process.RCTConfigProducers.eicIsolationThreshold = cms.uint32(7)
```
Comment (richard-cms) [inline L1Trigger/L1TCommon/python/customsPostLS1.py]: @mulhearn Two of the HLT breaks come from my commits - they can be fixed by adding "hasattr" guards on process.RCTConfigProducers here.

**Comment by mulhearn** [comment]
The remaining error is due to the new L1TCaloConfig DB table, which is intended to remove our dependence on config files.  In principle this could be fixed by a new GT, but this isn't ready yet.  Our intention was to default to local config but it seems something went wrong.  Let me remind myself how to default this table properly...

**Comment by mulhearn** [comment]
Until we have the new GT, the HLT workflows needs to call: 

   process.load('L1Trigger.L1TCalorimeter.caloConfigStage1PP_cfi')

**Comment by mulhearn** [comment]
This didn't break the workflows in runTheMatrix.py because it is being setup by "customiseSimL1EmulatorForStage1".  We clearly need to add some additional tests for our unit testing...

**Comment by mulhearn** [comment]
Actually looks like this is just a mistake in the 50ns customizations... it is not calling the setup common to 25/50 ns.

**Comment by mulhearn** [comment]
Seems to work...

**Pull request merged**